elasticsearch/connection/http_urllib3.py

```diff
@@ -85,16 +85,25 @@ def __init__(self, host='localhost', port=9200, http_auth=None,
 
         # if ssl_context provided use SSL by default
         if self.use_ssl or ssl_context:
-            ca_certs = CA_CERTS if ca_certs is None else ca_certs
-
-            if not ca_certs and not ssl_context and verify_certs:
-                # If no ca_certs and no sslcontext passed and asking to verify certs
-                # raise error
-                raise ImproperlyConfigured("Root certificates are missing for certificate "
-                    "validation. Either pass them in using the ca_certs parameter or "
-                    "install certifi to use it automatically.")
-            if verify_certs or ca_certs or ssl_version:
-                warnings.warn('Use of `verify_certs`, `ca_certs`, `ssl_version` have been deprecated in favor of using SSLContext`', DeprecationWarning)
+            if ca_certs:
+                warnings.warn('Use of `ca_certs` has been deprecated in favor of using SSLContext`', DeprecationWarning)
+            else:
+                ca_certs = CA_CERTS
+
+            if verify_certs:
+                warnings.warn('Use of `verify_certs` has been deprecated in favor of using SSLContext', DeprecationWarning)
+
+            if not ssl_context:
+                # if no SSLContext we must make one
+
+                if not ca_certs and verify_certs:
```

Comment:
ca_certs will only be Empty is certifi is not installed. Might be better if passed into the create_ssl_context function.